paperslicer/media/extractor.py

```
@@ -0,0 +1,135 @@
+import os
+import hashlib
+from typing import List, Dict
+
+import fitz  # PyMuPDF
+
+
+def _hash(s: str) -> str:
+    return hashlib.sha1(s.encode("utf-8")).hexdigest()[:8]
```

Ruff (0.12.2)

Probable use of insecure hash functions in `hashlib`: `sha1`

(S324)